content/geometry/SegmentIntersection.h

```diff
@@ -1,53 +1,46 @@
 /**
- * Author: Ulf Lundstrom
- * Date: 2009-03-21
+ * Author: Victor Lecomte, chilli
+ * Date: 2019-04-27
  * License: CC0
- * Source:
+ * Source: https://vlecomte.github.io/cp-geo.pdf
  * Description:\\
 \begin{minipage}{75mm}
-If a unique intersetion point between the line segments going from s1 to e1 and from s2 to e2 exists r1 is set to this point and 1 is returned.
-If no intersection point exists 0 is returned and if infinitely many exists 2 is returned and r1 and r2 are set to the two ends of the common line.
+If a unique intersection point between the line segments going from s1 to e1 and from s2 to e2 exists then it is returned.
+If no intersection point exists an empty vector is returned. If infinitely many exist a vector with 2 elements is returned.
 The wrong position will be returned if P is Point<int> and the intersection point does not have integer coordinates.
```

Comment:
`Point<int>` ->` Point<ll>` while you're changing things, no-one should have reason to use `Point<int>` anymore